core/src/main/protobuf/org/apache/spark/status/protobuf/store_types.proto

```diff
@@ -128,6 +128,8 @@ message TaskDataWrapper {
   int64 shuffle_merged_local_bytes_read = 49;
   int64 shuffle_remote_reqs_duration = 50;
   int64 shuffle_merged_remote_req_duration = 51;
+  int64 peak_on_heap_execution_memory = 52;
+  int64 peak_off_heap_execution_memory = 53;
```

Comment:
Add to `ExecutorStageSummary`, `ExecutorSummary`, `ExecutorMetricsDistributions` as well.
(Here and other model classes)

Reply:
Looks like we already have executor level stage memory metrics:
- `ExecutorStageSummary` has `val peakMemoryMetrics: Option[ExecutorMetrics]`
- `ExecutorSummary` has `val memoryMetrics: Option[MemoryMetrics]`
- `ExecutorMetricsDistributions` has `val peakMemoryMetrics: ExecutorPeakMetricsDistributions`

which are aggregated through [AppStatusListener.updateStageLevelPeakExecutorMetrics](https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/status/AppStatusListener.scala#L1000)